src/test/java/com/microsoft/azure/spark/tools/http/HttpObservableScenario.java

```diff
@@ -0,0 +1,204 @@
+import static org.junit.Assert.assertTrue;
+
+public class HttpObservableScenario {
+    private HttpObservable httpMock;
+    private MockHttpService httpServerMock = null;
+    private HttpRequestBase httpRequest;
+    private CookieStore cookieStore;
+    private MockHttpService httpsServerMock;
+    private boolean isMockTSInvoked = false;
+
+    @Before
+    public void setUp() {
+        this.httpServerMock = MockHttpService.create();
+        this.httpsServerMock = MockHttpService.createHttps();
+        this.cookieStore = new BasicCookieStore();
+    }
+
+    @Given("^setup a basic Http mock service for '(.+)' request '(.*)' to return '(.*)'")
+    public void setupMockService(String method, String path, String response) {
+        this.httpServerMock.stub(method, path, 200, response);
+    }
+
+    @Given("^setup a basic Https mock service for '(.+)' request '(.*)' to return '(.*)'")
+    public void setupMockHttpsService(String method, String path, String response) {
+        this.httpsServerMock.stubHttps(method, path, 200, response);
+    }
+
+    @Given("^prepare Http '(.+)' request to '(.*)' with username '(.+)' and password '(.+)'")
+    public void prepareRequest(String method, String path, String username, String password) throws Throwable {
+        this.httpMock = new HttpObservable(username, password);
+
+        this.httpRequest = new HttpRequestBase() {
+            @Override
+            public String getMethod() {
+                return method;
+            }
+        };
+
+        this.httpRequest.setURI(URI.create(httpServerMock.completeUrl(path)));
+    }
+
+    @Given("^prepare Https '(.+)' request to '(.*)' with username '(.+)' and password '(.+)'")
+    public void prepareHttpsRequest(String method, String path, String username, String password) throws Throwable {
+        this.httpMock = new HttpObservable(username, password);
+
+        this.httpRequest = new HttpRequestBase() {
+            @Override
+            public String getMethod() {
+                return method;
+            }
+        };
+
+        this.httpRequest.setURI(URI.create(httpsServerMock.completeHttpsUrl(path)));
+    }
+
+    @Given("^add a cookie '(.*):(.*)' into cookie store")
+    public void mockCookieStore(String name, String value) {
+        BasicClientCookie cookie = new BasicClientCookie(name, value);
+        cookie.setExpiryDate(DateTime.now().plus(Period.days(1)).toDate());
+        cookie.setDomain("localhost");
+        cookieStore.addCookie(cookie);
+
+        this.httpMock.setCookieStore(this.cookieStore);
+    }
+
+    @Then("^send and check Http '(.+)' request to '(.*)' should contain headers$")
+    public void checkRequestHeaderAndParameters(String method,
+                                                String path,
+                                                Map<String, String> expectHeaders) {
+        this.httpMock.request(this.httpRequest, null, null, null ).subscribe();
```

Comment:
Do we need to block this observable?

Reply:
No need, the current scheduler is in the current thread